src/TroykaButton.cpp

```diff
@@ -9,64 +9,85 @@
 
 #include "TroykaButton.h"
 
-TroykaButton::TroykaButton(uint8_t pin, uint32_t timeHold, bool pullUP) {
+TroykaButton::TroykaButton(uint8_t pin, uint32_t timeHold, bool pullUP, uint32_t repeatDelay) {
     _pin = pin;
     _timeHold = timeHold;
+    _repeatDelay = repeatDelay;
     _pullUP = pullUP;
 }
 
 // инициализация кнопки
 void TroykaButton::begin() {
-    pinMode(_pin, INPUT);
-    _msButtonState = 0;
-    _buttonStateOld = ON_RELEASE;
-	_isClick = false;
+    if (_pullUP)
+        pinMode(_pin, INPUT_PULLUP);
+    else
+        pinMode(_pin, INPUT);
```

Comment:
Так короче:

```suggestion
    pinMode(_pin, _pullUP ? INPUT_PULLUP : INPUT);
```

Правильно понимаю, что раньше функциональность была заявлена, но фактически настройка `pullUP` игнорировалась?

Reply:
да